app/domain/authentication/authn_jwt/signing_key/signing_key_settings_builder.rb

```diff
@@ -0,0 +1,131 @@
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # fetches signing key settings, validates and builds SigningKeysSettings object
+      SigningKeySettingsBuilder = CommandClass.new(
+        dependencies: {
+          signing_key_settings_class: Authentication::AuthnJwt::SigningKey::SigningKeySettings
+        },
+        inputs: %i[signing_key_parameters]
+      ) do
+        def call
+          validate_signing_key_parameters
+          signing_key_settings
+        end
+
+        private
+
+        def validate_signing_key_parameters
+          single_signing_key_source
+          cert_store_only_with_jwks_uri
+          public_keys_have_issuer
+        end
+
+        def single_signing_key_source
+          check_no_signing_keys_source
+          check_all_signing_keys_sources
+          check_jwks_provider_uri_pair
+          check_jwks_uri_public_keys_pair
+          check_public_keys_provider_uri_pair
+        end
+
+        def check_no_signing_keys_source
+          return unless !jwks_uri && !provider_uri && !public_keys
+
+          raise Errors::Authentication::AuthnJwt::InvalidSigningKeySettings,
+                "One of #{JWKS_URI_RESOURCE_NAME}, #{PUBLIC_KEYS_RESOURCE_NAME}, and #{PROVIDER_URI_RESOURCE_NAME} have to be defined"
+        end
+
+        def check_all_signing_keys_sources
+          return unless jwks_uri && public_keys &&  provider_uri
+
+          raise Errors::Authentication::AuthnJwt::InvalidSigningKeySettings,
+                "#{JWKS_URI_RESOURCE_NAME}, #{PUBLIC_KEYS_RESOURCE_NAME}, and #{PROVIDER_URI_RESOURCE_NAME} cannot be define simultaneously"
+        end
+
+        def check_jwks_provider_uri_pair
+          return unless jwks_uri && provider_uri
+
+          raise Errors::Authentication::AuthnJwt::InvalidSigningKeySettings,
+                "#{JWKS_URI_RESOURCE_NAME} and #{PROVIDER_URI_RESOURCE_NAME} cannot be define simultaneously"
```

Comment:
Cant we move this template to errors.rb?

Reply:
Actually errors.rb contains only `TrackableErrorClass` instances so moving plain text messages to it feels me not right.
But I'll move those texts to consts of the same class